### PR TITLE
fix: Add Touche2020v3 and JMTEB

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist

- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: #749 #1170 

- [X] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [X] `facebook/contriver`
    - [Touche2020v3.json](https://github.com/user-attachments/files/17180881/Touche2020v3.json)
  - [X] `intfloat/multilingual-e5-small`
    [JMTEB leaderboard to compare](https://github.com/sbintuitions/JMTEB/blob/main/leaderboard.md). I ran `MrTidyRetrieval` only for Bengali, as the Japanese version has 7M documents in context to encode. Overall, the results are similar, with the main difference caused by that JMTEB doesn't include titles.
    - [MrTidyRetrieval.json](https://github.com/user-attachments/files/17180876/MrTidyRetrieval.json)
    - [ESCIReranking.json](https://github.com/user-attachments/files/17180877/ESCIReranking.json)
    - [JaqketRetrieval.json](https://github.com/user-attachments/files/17180878/JaqketRetrieval.json)
- [X] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [X] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/tasks/Retrieval/eng/Touche2020Retrieval.py [hunk 1]
```diff
@@ -57,3 +56,44 @@ class Touche2020(AbsTaskRetrieval):
             },
         },
     )
+
+
+class Touche2020v3(AbsTaskRetrieval):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/Touche2020Retrieval.py:61]:
```suggestion
class Touche2020v3Retrieval(AbsTaskRetrieval):
```

File: mteb/tasks/Retrieval/eng/Touche2020Retrieval.py [hunk 2]
```diff
@@ -57,3 +56,44 @@ class Touche2020(AbsTaskRetrieval):
...
+    metadata = TaskMetadata(
+        name="Touche2020v3",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/Touche2020Retrieval.py:63]:
```suggestion
        name="Touche2020v3Retrieval",
```
To differentiate from the alternative: "Touche2020v3Retrieval.v3" (which signifies a version change with MTEB).

File: mteb/tasks/Retrieval/eng/Touche2020Retrieval.py [hunk 3]
```diff
@@ -1,9 +1,8 @@
 from __future__ import annotations
 
+from mteb.abstasks.AbsTaskRetrieval import AbsTaskRetrieval
 from mteb.abstasks.TaskMetadata import TaskMetadata
 
-from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
-
 
 class Touche2020(AbsTaskRetrieval):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/Touche2020Retrieval.py:7]: Should we add a `superseded_by` to this datasets

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
Wonderful thanks for doing this!

We should also add JMTEB to the benchmarks.py file

**Review by KennethEnevoldsen — APPROVED** [review]

**Comment by Samoed** [comment]
@KennethEnevoldsen runner hanged up

**Pull request merged**